File: 151004/Bosko/lab2/lab2/DTO/UserRequestTo.cs
using lab2.DTO.Interface;

namespace lab2.DTO
{
    public class UserRequestTo(int Id, string? Login, string? Password, string? Firstname, string? Lastname) : IRequestTo
    {
        public int Id { get; set; } = Id;

        public string? Login { get; set; } = Login;

        public string? Password { get; set; } = Password;

        public string? Firstname { get; set; } = Firstname;

        public string? Lastname { get; set; } = Lastname;
    }
}

File: 151004/Gonzarevich/lab4/publisher/DTO/UserRequestTo.php
<?php

namespace App\DTO;

use Symfony\Component\Validator\Constraints as Assert;

class UserRequestTo
{
    public function __construct(

        public readonly ?int $id,

        #[Assert\NotBlank]
        #[Assert\Length(min: 2, max: 64, minMessage: 'Login minimal length is {{ limit }} chars.')]
        public readonly string $login,

        #[Assert\NotBlank]
        #[Assert\Length(min: 8, max: 128)]

        #[Assert\UserPassword]
        public readonly string $password,

        #[Assert\NotBlank]
        #[Assert\Length(min: 2, max: 64)]
        public readonly string $firstname,
        
        #[Assert\NotBlank]
        #[Assert\Length(min: 2, max: 64)]
        public readonly string $lastname,
    ) {
    }
}